src/Files.App/ViewModels/Properties/CustomizationViewModel.cs

```diff
@@ -66,40 +69,32 @@ public CustomizationViewModel(IShellPage appInstance, BaseProperties basePropert
 			// Get default
 			LoadIconsForPath(IconResourceItemPath);
 
-			RestoreDefaultIconCommand = new RelayCommand(ExecuteRestoreDefaultIcon);
-			OpenFilePickerCommand = new AsyncRelayCommand(ExecuteOpenFilePickerAsync);
+			RestoreDefaultIconCommand = new RelayCommand(ExecuteRestoreDefaultIconCommand);
+			OpenFilePickerCommand = new RelayCommand(ExecuteOpenFilePickerCommand);
 		}
 
-		private void ExecuteRestoreDefaultIcon()
+		private void ExecuteRestoreDefaultIconCommand()
 		{
 			SelectedDllIcon = null;
 			_isIconChanged = true;
 		}
 
-		private async Task ExecuteOpenFilePickerAsync()
+		private void ExecuteOpenFilePickerCommand()
 		{
-			// Initialize picker
-			FileOpenPicker picker = new()
-			{
-				SuggestedStartLocation = PickerLocationId.ComputerFolder,
-				ViewMode = PickerViewMode.Thumbnail,
-			};
-
-			picker.FileTypeFilter.Add(".dll");
-			picker.FileTypeFilter.Add(".exe");
-			picker.FileTypeFilter.Add(".ico");
-
-			// WINUI3: Create and initialize new window
 			var parentWindowId = _appWindow.Id;
-			var handle = Microsoft.UI.Win32Interop.GetWindowFromWindowId(parentWindowId);
-			WinRT.Interop.InitializeWithWindow.Initialize(picker, handle);
+			var hWnd = Microsoft.UI.Win32Interop.GetWindowFromWindowId(parentWindowId);
 
-			// Open picker
-			var file = await picker.PickSingleFileAsync();
-			if (file is null)
-				return;
+			string[] extensions =
+			[
+				"Application extension", "*.dll",
```

Comment:
Should the text be localized?

Reply:
I think yes and no, as wondering if we can fetch this from File Explorer because this text is corresponding to FE’s friendly extension name.

Reply:
Localizing should be easy, but if you want, we can hold off displaying this text and add it in another PR.

Reply:
I couldn't find any API related to friendly file extension, this maybe comes from registry?
Now, I think I'd like to localize.

Reply:
I think it's usually handled by whichever app is using the dialog.

Reply:
Done localizing